client/ayon_silhouette/api/lib.py

```diff
@@ -249,3 +251,41 @@ def reset_session_settings(session=None, task_entity=None):
     with undo_chunk("Reset session settings"):
         set_resolution_from_entity(session, task_entity)
         set_frame_range_from_entity(session, task_entity)
+
+
+@contextlib.contextmanager
+def capture_messageboxes(callback):
+    """Capture messageboxes and call a callback with them.
+
+    This is a workaround for Silhouette not allowing the Python code to
+    suppress messageboxes and supply default answers to them. So instead we
+    capture the messageboxes and respond to them through a rapid QTimer.
+    """
+    processed = set()
+    timer = QtCore.QTimer()
+
+    def on_timeout():
+        # Check for dialogs
+        widgets = QtWidgets.QApplication.instance().topLevelWidgets()
+        has_boxes = False
+        for widget in widgets:
+            if isinstance(widget, QtWidgets.QMessageBox):
+                has_boxes = True
+                if widget in processed:
+                    continue
+                processed.add(widget)
+                callback(widget)
+        if not has_boxes:
```

Comment:
This would stop the capture if the dialog is not yet shown, or?

Reply:
It would - yes, but I've been unable to hit such a case with the export methods of Silhouette. But maybe it's 'redundant' stopping early.

Reply:
Just asking, if it works, not a problem (maybe small comment for future generations?).

Reply:
https://github.com/ynput/ayon-silhouette/pull/10/commits/e6adaa841d06b267601523ca30a8dc9ce2c36112